### PR TITLE
DOC: conversion to numpy docstrings

### PR DESCRIPTION
**Pull request by klauer**
See something; say something (please).

@tacaswell, I'm sure any major snafus will be very obvious to you, so give it a look if you have a chance

### CODE REVIEW

File: ophyd/controls/areadetector/detectors.py [hunk 1]
```diff
@@ -1005,9 +1023,8 @@ def remove_macros(record):
     print('class %s(%s):' % (det_name, base_class.__name__))
 
     def get_prop_name(pv):
-        '''
-        A terribly confusing method to get a property name from
-        the camel-case AreaDetector PV names
+        '''A terribly confusing method to get a property name from the
+        camel-case AreaDetector PV names
```
Comment (tacaswell) [inline ophyd/controls/areadetector/detectors.py]: This docstring does not inspire confidence in the function it documents.

File: ophyd/controls/areadetector/plugins.py [hunk 2]
```diff
@@ -713,15 +729,20 @@ def plugin_from_pvname(pv):
...
     the prefix, suffix, and any kwargs for the constructor.
 
     Uses `plugin_from_pvname` first, but falls back on using
     epics channel access to determine the plugin type.
 
-    :returns: plugin instance
-    :raises: ValueError if the plugin type can't be determined
+    Returns
+    -------
+        plugin instance
```
Comment (tacaswell) [inline ophyd/controls/areadetector/plugins.py]: this looks a bit funny

File: ophyd/controls/areadetector/plugins.py [hunk 3]
```diff
@@ -738,15 +759,20 @@ def get_areadetector_plugin_class(prefix, suffix=''):
...
```
Comment (tacaswell) [inline ophyd/controls/areadetector/plugins.py]: ditto

File: ophyd/controls/cas/motor.py [hunk 4]
```diff
@@ -35,6 +35,21 @@
...
+
+    Parameters
+    ----------
+    name : str
+        The record name (not including the server prefix)
+    positioner : Positioner
+        The ophyd :class:`Positioner` to expose to EPICS
+    tweak_value : float
+        The default tweak value
+    **kwargs : dict
```
Comment (tacaswell) [inline ophyd/controls/cas/motor.py]: typically don't document kwargs

Comment (tacaswell) [inline ophyd/controls/cas/motor.py]: Or, rather it typically goes in the description like "kwargs are passed through to base-class, see CasRecord for possible keys"

File: ophyd/controls/cas/pv.py [hunk 5]
```diff
@@ -616,20 +613,33 @@ def __repr__(self):
...
+    Starts out with just a VAL field. Additional fields can be added
+    dynamically.
+
+    Parameters
+    ----------
+    name : str
+        The record prefix
+    val_field
+        The default value for the value field
+    rtype : str
```
Comment (tacaswell) [inline ophyd/controls/cas/pv.py]: these are missing ', optional'

File: ophyd/controls/scaler.py [hunk 6]
```diff
@@ -114,11 +117,16 @@ def preset_time(self, time):
...
+        Parameters
+        ----------
+        channels
+            a tuple enumerating the channels to return.
 
-        :param channels: a tuple enumerating the channels to return.
-        :returns: a dict {channel x: counts,}
+        Returns
+        -------
+        {channel_x: counts}
```
Comment (tacaswell) [inline ophyd/controls/scaler.py]: This looks funny

File: ophyd/controls/signal.py [hunk 7]
```diff
@@ -138,6 +135,33 @@ def read(self):
 
 
 class EpicsSignal(Signal):
+    '''An EPICS signal, comprised of either one or two EPICS PVs
+
+    =================================
```
Comment (tacaswell) [inline ophyd/controls/signal.py]: I am not 100% that this is a well formed rst table

File: ophyd/utils/epics_pvs.py [hunk 8]
```diff
@@ -29,11 +28,16 @@
...
+    Parameters
+    ----------
+    pv : str
+        the pv to split
 
-    :param str pv: the pv to split
-    :returns: (record, field)
+    Returns
+    -------
+    (record, field)
```
Comment (tacaswell) [inline ophyd/utils/epics_pvs.py]: This isn't quite right, I think just drop the '()'

File: ophyd/utils/epics_pvs.py [hunk 9]
```diff
@@ -200,14 +217,17 @@ def waveform_to_string(value, type_=str, delim=''):
...
+    Due to a bug in certain versions of PyEpics, form='time' cannot be used
+    with some large arrays.
 
     native: gives time.time() timestamps from this machine
     time: gives timestamps from the PVs themselves
 
-    :returns: 'native' or 'time'
+    Returns
+    -------
+    'native' or 'time'
```
Comment (tacaswell) [inline ophyd/utils/epics_pvs.py]: I would write this as `{'native', 'time'}`.  Only really important if you want to expose this to vistrails.

File: ophyd/utils/epics_pvs.py [hunk 10]
```diff
@@ -246,10 +266,11 @@ def _naive_parse_version(version):
...
 
 def records_from_db(fn):
-    '''
-    Naively parses db/template files looking for record names
+    '''Naively parse db/template files looking for record names
 
-    :returns: [(record type, record name), ...]
+    Returns
+    -------
+    [(record type, record name), ...]
```
Comment (tacaswell) [inline ophyd/utils/epics_pvs.py]: maybe do this as 

```
ret : list
    [(record type, record name), ...]
```

**Comment by tacaswell** [comment]
I did a skim and found a few things, but nothing egregious.

**Comment by tacaswell** [comment]
I really hope you used an automated tool to do this...

**Comment by klauer** [comment]
Thanks, @tacaswell - that will help things along. I think 82197f8 should address all of your concerns.

Yes, wrote a quick script to automate the bulk of it (though being lax with the original docstrings left some things to be fixed up).

File: ophyd/controls/areadetector/detectors.py (hunk 1 above)
Comment (stuwilkins) [inline ophyd/controls/areadetector/detectors.py]: LOL: It also has areaDetector which also doesn’t inspire confidence …… :-)

On Jan 27, 2015, at 6:24 PM, Thomas A Caswell <notifications@github.com<mailto:notifications@github.com>> wrote:


In ophyd/controls/areadetector/detectors.py<https://github.com/NSLS-II/ophyd/pull/53#discussion_r23653107>:

> @@ -1005,9 +1023,8 @@ def remove_macros(record):
>      print('class %s(%s):' % (det_name, base_class.__name__))
>
>      def get_prop_name(pv):
> -        '''
> -        A terribly confusing method to get a property name from
> -        the camel-case AreaDetector PV names
> +        '''A terribly confusing method to get a property name from the
> +        camel-case AreaDetector PV names


This docstring does not inspire confidence in the function it documents.

Comment (klauer) [inline ophyd/controls/areadetector/detectors.py]: I can't argue with either point!

**Pull request merged**